internal/edgedbtypes/bool.go

```diff
@@ -37,3 +41,23 @@ func (o *OptionalBool) Unset() {
 	o.val = false
 	o.isSet = false
 }
+
+func (o OptionalBool) MarshalJSON() ([]byte, error) {
+	if o.isSet {
+		return json.Marshal(o.val)
+	}
+	return json.Marshal(nil)
+}
+
+func (o *OptionalBool) UnmarshalJSON(bytes []byte) error {
+	if string(bytes) == "null" {
```

Comment:
I take that back. There doesn't seem to be much of a difference here.

Reply:
According to the docs (https://pkg.go.dev/encoding/json#Unmarshaler):

> The input can be assumed to be a valid encoding of a JSON value.

So I think it's safe here to just check that the first byte is `n`, since `null` is the only valid JSON value that can start with `n`, which I think is the most efficient check here.